### PR TITLE
Setup CI for v2

### LINKED ISSUE #2883: Setup CI for v2
## Requirement - what kind of business use case are you trying to solve?

We need to set up a custom CI infra for the v2, as there are things from v1 that we might not need, do differently, or that don't exist yet.

## Proposal - what do you suggest to solve the problem or improve the existing situation?

- [ ] Set the current CI to ignore PRs that include only changes to v2
- [ ] Create a new Makefile inside the v2 with the targets we need for v2
- [ ] Create a new v2-specific GitHub workflows

### PR DESCRIPTION
**Pull request by jpkrohling**
## Short description of the changes
- Moved v1-specific workflows to .github/workflows/v1/
- Makefile for v1 should ignore v2 files (linting, header checks, ...)
- Add the manifests for the agent
- Add builder configuration to build the agent
- Add goreleaser to perform a release when new v2 tags are created

Fixes #2883

### CODE REVIEW

File: v2/storage/memory/config.go [hunk 1]
```diff
@@ -1,10 +1,10 @@
-// Copyright (c) 2021 The Jaeger Authors.
```
Comment (albertteoh) [inline v2/storage/memory/config.go:1]: Why remove the copyright year?

Comment (jpkrohling) [reply v2/storage/memory/config.go:1]: My understanding from other communities is that it isn't really required. And not having it makes it less cumbersome when the year changes.

File: v2/Makefile [hunk 2]
```diff
@@ -0,0 +1,54 @@
+VERSION ?= 2.0.0-apha2
```
Comment (Ashmita152) [inline v2/Makefile:1]: Is it a typo in alpha.

File: v2/.goreleaser.yaml [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+    binary: jaeger-agent
+    flags: 
+      - -trimpath
+    gcflags:
+      - all=-N -l
+    env:
+      - CGO_ENABLED=0
+    goos:
+      - linux
+      - darwin
```
Comment (Ashmita152) [inline v2/.goreleaser.yaml:24]: Do we need to support windows in v2 as well as we do in v1 ?

File: v2/Makefile (hunk 2 above)
Comment (jpkrohling) [reply v2/Makefile:1]: Great catch!

File: v2/.goreleaser.yaml (hunk 3 above)
Comment (jpkrohling) [reply v2/.goreleaser.yaml:24]: Yes. I temporarily removed as I hit an error and forgot to put it back. Thanks for spotting it!

File: v2/.goreleaser.yaml [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+  ## 1. regular, production optimized
+  ## 2. debug, with no optimization or inlining
+  - &agent-build
+    id: jaeger-agent
+    dir: ./_build/agent
+    binary: jaeger-agent
+    flags: 
+      - -trimpath
+    gcflags:
+      - all=-N -l
```
Comment (Ashmita152) [inline v2/.goreleaser.yaml:19]: Does this means we are turning off compiler optimizations in normal image build too ?

Comment (jpkrohling) [reply v2/.goreleaser.yaml]: Oops!

**Pull request closed without merging**